From 320993a2c69e833824da09956573c366803410db Mon Sep 17 00:00:00 2001
From: Ramon Quitales <ramon@pulumi.com>
Date: Thu, 18 May 2023 15:21:33 -0700
Subject: [PATCH 26/29] Fix spurrious json diff for redrive_policy

We need to normalize the json input to compare agasint the one stored
in state.

Fix for https://github.com/pulumi/pulumi-aws/issues/2307.
---
 internal/service/sqs/queue_redrive_policy.go | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/internal/service/sqs/queue_redrive_policy.go b/internal/service/sqs/queue_redrive_policy.go
index 7f55a8e6e8..d3a620f555 100644
--- a/internal/service/sqs/queue_redrive_policy.go
+++ b/internal/service/sqs/queue_redrive_policy.go
@@ -8,6 +8,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/structure"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/verify"
 )
 
 // @SDKResource("aws_sqs_queue_redrive_policy")
@@ -31,9 +32,10 @@ func ResourceQueueRedrivePolicy() *schema.Resource {
 				ForceNew: true,
 			},
 			"redrive_policy": {
-				Type:         schema.TypeString,
-				Required:     true,
-				ValidateFunc: validation.StringIsJSON,
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateFunc:     validation.StringIsJSON,
+				DiffSuppressFunc: verify.SuppressEquivalentJSONDiffs,
 				StateFunc: func(v interface{}) string {
 					json, _ := structure.NormalizeJsonString(v)
 					return json
-- 
2.41.0

